stack_test.go

```diff
@@ -4,19 +4,20 @@ import (
 	"fmt"
 	"testing"
 
+	"strings"
+
 	"github.com/stretchr/testify/assert"
 )
 
 func TestCaller(t *testing.T) {
 	s := NewStackTrace(0)
-	assert.NotEmpty(t, s.StackTrace)
+	assert.Len(t, s.StackTrace, 3)
```

Comment:
I think asserting that the length is 3 here breaks this test, for two reasons:

1) I don't know if for other archs the count is still 3, it may be more if more files are involved in this computation.

2) this will break if the Golang codebase is modified in the future adding or removing files involved in this execution.

I think a better solution for all the things we have been trying out is to call `debug.Stack` before line 13, then use its output to build an "expected" value for the test that is independent of the arch and the names of the source files in the golang project. For instance, something like this is a good start:
```
	var buf bytes.Buffer
	buf.Write(debug.Stack())
	expected := getFileNamesFromStack(buf)x
```
Writing `getFileNamesFromStack should not be difficult, just print `buf.String()` to check out the format, which is quite simple and arch independent.